File: workers/src/utils/types/enums/eCurrency.ts
export enum eCurrency {
  USD = 'USD',
  EUR = 'EUR',
}

export function getECurrencyFromString(str: string): eCurrency | null {
  switch (str.toUpperCase()) {
    case 'USD':
      return eCurrency.USD;
    case 'EUR':
      return eCurrency.EUR;
    default:
      return null;
  }
}

export function getECurrencyBySymbol(text: string): eCurrency | null {
  if (text.includes('$')) return eCurrency.USD;
  if (text.includes('€')) return eCurrency.EUR;
  return null;
}

File: src/GamesFinder.Orchestrator.Domain/Enums/ECurrency.cs
namespace GamesFinder.Orchestrator.Domain.Enums;

public enum ECurrency
{
  EUR,
  USD
}

public static class ECurrencyHelpers {
  public static ECurrency? GetECurrency(string currency)
  {
    switch (currency)
    {
      case "EUR":
      case "eur":
      case "Eur":
        return ECurrency.EUR;
      case "USD":
      case "Usd":
      case "usd":
        return ECurrency.USD;
      default:
        return null;
    }
  }
}
